server/app/models/account.rb

```diff
@@ -1,24 +1,27 @@
 class Account < ApplicationRecord
   has_many :account_books
   has_many :ckb_transactions, through: :account_books
+  validates_presence_of :address_hash, :balance, :cell_consumed, :ckb_transactions_count
+  validates :balance, :cell_consumed, :ckb_transactions_count, numericality: { greater_than_or_equal_to: 0 }
 
   def self.find_or_create_account(ckb_transaction, verify_script)
     address_hash = CKB::Utils.json_script_to_type_hash(verify_script)
-    if Account.where(address_hash: [address_hash[2..-1]].pack("H*")).exists?
-      account = Account.find_by(address_hash: [address_hash[2..-1]].pack("H*"))
+    if Account.where(address_hash: [address_hash.delete_prefix(ENV["DEFAULT_HASH_PREFIX"])].pack("H*")).exists?
```

Comment:
This `if`/`else` could be:

```ruby
account = Account.find_by(address_hash: [address_hash.delete_prefix(ENV["DEFAULT_HASH_PREFIX"])].pack("H*"))
account ||= Account.create(address_hash: address_hash, balance: 0, cell_consumed: 0)
```

Reply:
I think `exists ` should be faster when the amount of data is large.

Reply:
You still need to load account if exists, so check exists it no need, `exists?` using `SELECT 1 as one` to avoid IO read operation, so it can faster than `find_by` if record exists. In this situation, it will be even slower for redundant SQL